server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -85,8 +85,9 @@ private static Throwable deduplicateCause(Throwable cause, ShardSearchFailure[]
     @Override
     public RestStatus status() {
         if (shardFailures.length == 0) {
-            // if no successful shards, it means no active shards, so just return SERVICE_UNAVAILABLE
-            return RestStatus.SERVICE_UNAVAILABLE;
+            // if no successful shards, the failure could have happened even before the fetch phase started
+            // so try to get the status from the cause instead of returning SERVICE_UNAVAILABLE blindly
```

Comment:
shall we explicitly mention rejection from the fetch phase given that this is the problem, and it does not have to do with timing?

Reply:
@javanna I have updated the comment to explicitly mention rejected exception